weldx/transformations.py

```diff
@@ -2772,6 +2877,89 @@ def plot(self, ax=None):
             graph, pos, ax, with_labels=True, font_weight="bold", node_color=color_map
         )
 
+    def plot_coordinate_systems(
+        self,
+        backend="mpl",
+        axes=None,
+        reference_system=None,
+        coordinate_systems=None,
+        colors=None,
+        title=None,
+        limits=None,
+        time=None,
+        time_ref=None,
+        show_labels=True,
+        show_origins=True,
+        show_traces=True,
+        show_vectors=True,
+    ):
+        """Plot the coordinate systems of the coordinate system manager.
+
+        Parameters
+        ----------
+        backend : str
+            Select the rendering backend of the plot. The options are:
+
+            - `k3d` to get an interactive plot using [k3d](https://k3d-jupyter.org/)
+            - `mpl` for static plots using [matplotlib](https://matplotlib.org/)
+
+            Note that k3d only works inside jupyter notebooks
+        axes :
+            (matplotlib only) The target axes object that should be drawn to. If `None`
+            is provided, a new one will be created.
+        reference_system : str
+            (matplotlib only) The name of the reference system for the plotted
+            coordinate systems
+        time : pandas.DatetimeIndex, pandas.TimedeltaIndex, List[pandas.Timestamp], or \
+               LocalCoordinateSystem
+            The time steps that should be plotted
+        time_ref : pandas.Timestamp
+            A reference timestamp that can be provided if the ``time`` parameter is a
+            `pandas.TimedeltaIndex`
+        show_traces : bool
+            (matplotlib only) If `True`, the trace of time dependent coordinate systems
+            is plotted.
+        show_vectors :
+            (matplotlib only) If `True`, the coordinate cross of time dependent
+            coordinate systems is plotted.
+
+        """
+        if backend == "k3d":
+            CoordinateSystemManagerVisualizerK3D(
+                csm=self,
+                reference_system=reference_system,
+                coordinate_systems=coordinate_systems,
+                colors=colors,
+                title=title,
+                limits=limits,
+                time=time,
+                time_ref=time_ref,
+                show_labels=show_labels,
+                show_origins=show_origins,
+                show_traces=show_traces,
+                show_vectors=show_vectors,
+            )
+        elif backend == "mpl":
+            set_ax_eq = axes is None
```

Comment:
I would avoid setting axes=equal by default as much as possible, it's good for small examples but messes with visibility for the welding applications

Reply:
Well, this is a complicated topic. I see your point, but a simple fix is just setting new limits to the axes (forgot to return it but one can use `gca`). If we don't use equal axes, the coordinate systems get skewed with the current implementation, which looks strange.  So at the moment, we have the choice between: 

- default equal axes and adjust them to larger objects later
- default adjusted size with skewed coordinate systems and call `axis_equal` later

However, I have some ideas on how we can tackle this a bit better, but I think the most convenient fix, for now, would be to add another function parameter to enable/disable `axis_equal`. 

I am curious if k3d provides the possibility to distort a specific axis... currently, all plots have equal axes.